GenericsHomework/Node.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace GenericsHomework;
+
+public class Node<T>(T data)
+{
+    //Can add contraint such as "where T : class" to ensure homogeneous values
+    //Can possibly also add "public T Data { get; }" to ensure pproperty is non nullable.
+    public T Data { get; } = data;
+    private Node<T>? next; 
+    public Node<T> Next
+    {
+        get => next ?? this;
+        private set => next = value;
+    }
```

Comment:
```suggestion
public class Node<T>
{
    //Can add contraint such as "where T : class" to ensure homogeneous values
    //Can possibly also add "public T Data { get; }" to ensure pproperty is non nullable.
    public T Data { get; };
    public Node<T> Next {get; private set;}
    public Node(T data){
            Next = this;
            Data = data;
    }
```
Having a backing field is unnecessary and overcomplicated things. Just use a regular constructor instead of primary which will allow you to set Next to "this" instead of having a private backing field.